x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -76,6 +76,13 @@ func (ms msgServer) InsertWorkerPayload(ctx context.Context, msg *types.MsgInser
 
 	if msg.WorkerDataBundle.InferenceForecastsBundle.Inference != nil {
 		inference := msg.WorkerDataBundle.InferenceForecastsBundle.Inference
+		isInfererRegistered, err := ms.k.IsWorkerRegisteredInTopic(ctx, topicId, inference.Inferer)
+		if err != nil {
+			return nil, errorsmod.Wrapf(err, "Error address is not registered in this topic")
+		}
+		if !isInfererRegistered {
+			return nil, errorsmod.Wrapf(err, "Error address is not registered in this topic")
+		}
 		err = ms.k.AppendInference(ctx, topicId, *nonce, inference)
```

Comment:
for each actor (reputer, inferer, forecaster) we need to check if they already submitted data. doesn't matter at this time if we only take their first or latest response (assume all conditions for inclusion are still met), but we must be consistent with this policy

and certainly can't accept multiple bundles from the same actor for the same role at the time epoch

Reply:
yes, I will overwrite previous data